Mathlib/RingTheory/Ideal/NatInt.lean

```diff
@@ -0,0 +1,84 @@
+/-
+Copyright (c) 2025 Junyan Xu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Junyan Xu
+-/
+import Mathlib.Algebra.Order.Ring.Int
+import Mathlib.Data.Nat.Prime.Int
+import Mathlib.RingTheory.LocalRing.MaximalIdeal.Basic
+
+/-!
+# Prime ideals in ℕ and ℤ
+
+## Main results
+
+* `Ideal.isPrime_nat_iff`: the prime ideals in ℕ are ⟨0⟩, ⟨p⟩ (for prime `p`), and ⟨2, 3⟩ = {1}ᶜ.
+  The proof follows https://math.stackexchange.com/a/4224486.
+
+* `Ideal.isPrime_int_iff` : the prime ideals in ℤ are ⟨0⟩ and ⟨p⟩ (for prime `p`).
+-/
+
+/-- The natural numbers form a local semiring. -/
+instance : IsLocalRing ℕ where
+  isUnit_or_isUnit_of_add_one {a b} hab := by
+    have h : a = 1 ∨ b = 1 := by omega
+    apply h.imp <;> simp +contextual
+
+open IsLocalRing Ideal
+
+theorem Nat.mem_maximalIdeal_iff {n : ℕ} : n ∈ maximalIdeal ℕ ↔ n ≠ 1 := by simp
+
+theorem Nat.coe_maximalIdeal : (maximalIdeal ℕ : Set ℕ) = {1}ᶜ := by ext; simp
+
+theorem Nat.maximalIdeal_eq_span_two_three : maximalIdeal ℕ = .span {2, 3} := by
+  refine le_antisymm (fun n h ↦ ?_) (span_le.mpr <| Set.pair_subset (by simp) (by simp))
+  obtain lt | lt := (mem_maximalIdeal_iff.mp h).lt_or_gt
+  · obtain rfl := lt_one_iff.mp lt; exact zero_mem _
+  exact mem_span_pair.mpr <|
+    exists_add_mul_eq_of_gcd_dvd_of_mul_pred_le 2 3 n (by simp) (show 2 ≤ n by omega)
+
+theorem Ideal.isPrime_nat_iff {P : Ideal ℕ} :
```

Comment:
I'm amused to observe that `Nat` has Krull dimension 2.

Reply:
added this result